frontend/lib/src/components/core/Block/ElementNodeRenderer.tsx

```diff
@@ -484,6 +476,22 @@ const RawElementNodeRenderer = (
       )
     }
 
+    case "arrowVegaLiteChart":
+      const vegaLiteElement = node.vegaLiteChartElement as VegaLiteChartElement
+      return (
+        <ArrowVegaLiteChart
+          element={vegaLiteElement}
+          // Vega-lite chart can be used as a widget (when selections are activated) or
+          // an element. We only want to set the key in case of it being used as a widget.
+          // For the non-widget usage, the id will
+          // be undefined.
+          {...(vegaLiteElement.id && {
+            key: vegaLiteElement.id,
+          })}
```

Comment:
I would move this up to `const key = vegaLiteElement.id`
and then just set the key here for readability.

Reply:
I simplified it to: `key={vegaLiteElement.id || undefined}`